uefi/src/table/runtime.rs

```diff
@@ -582,6 +582,68 @@ impl Display for Time {
     }
 }
 
+impl TryFrom<&[u8]> for Time {
+    type Error = TimeError;
+
+    fn try_from(bytes: &[u8]) -> core::result::Result<Self, Self::Error> {
+        if size_of::<Time>() <= bytes.len() {
+            let (year, bytes) = bytes.split_at(size_of::<u16>());
+            // The byte slice is at least as large as [`Time`], so this cannot panic.
+            let year = u16::from_le_bytes(year.try_into().unwrap());
+
+            let (month, bytes) = bytes.split_at(size_of::<u8>());
+            let month = u8::from_le_bytes(month.try_into().unwrap());
+
+            let (day, bytes) = bytes.split_at(size_of::<u8>());
+            let day = u8::from_le_bytes(day.try_into().unwrap());
+
+            let (hour, bytes) = bytes.split_at(size_of::<u8>());
+            let hour = u8::from_le_bytes(hour.try_into().unwrap());
+
+            let (minute, bytes) = bytes.split_at(size_of::<u8>());
+            let minute = u8::from_le_bytes(minute.try_into().unwrap());
+
+            let (second, bytes) = bytes.split_at(size_of::<u8>());
+            let second = u8::from_le_bytes(second.try_into().unwrap());
```

Comment:
I think it would be a little clearer/shorter to directly access bytes at an offset rather than splitting each time:
```rust
            let year = u16::from_le_bytes(bytes[0..2].try_into().unwrap());
            let month = bytes[2];
            let day = bytes[3];
            [...]
```